### PR TITLE
JP-3527 Update to docs for IFU outlier detection

### LINKED ISSUE #8250: Update IFU outlier detection in readthedocs
_Issue [JP-3527](https://jira.stsci.edu/browse/JP-3527) was created on JIRA by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw):_

IFU outlier detection documentation in readthedocs is out of date, and still refers to the old method of blotting cubes.

### PR DESCRIPTION
**Pull request by jemorrison**
Resolves [JP-3527](https://jira.stsci.edu/browse/JP-3527)

Closes #8250

This PR updates the IFU outlier docs. I also updated some comments in the ifu_outlier code to make things more clear.

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [x] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -61,6 +61,8 @@ documentation
 - Reorganized ``jump`` and ``ramp_fitting`` step docs content that's split between
   the jwst and stcal repos. [#8253]
 
+- Updated ``outlier_detection`` for IFU data to explain the method clearer. [#8360]
```
Comment (hbushouse) [inline CHANGES.rst:64]:
```suggestion
- Updated ``outlier_detection`` for IFU data to explain the method more clearly. [#8360]
```

File: docs/jwst/outlier_detection/outlier_detection_ifu.rst [hunk 2]
```diff
@@ -14,7 +14,9 @@ in the :ref:`calwebb_detector1 <calwebb_detector1>` pipeline.
 At this time it is believed that NIRSpec IFU data also have bad pixels that vary with time, though the time variation is
 still under study.
 
-An algorithm was developed to flag pixels that are  outliers when compared to their neighbors for a set of
+The basic of that the idea of the outlier detection plan for IFU data  is to look for pixels on the detector
```
Comment (hbushouse) [inline docs/jwst/outlier_detection/outlier_detection_ifu.rst:17]: "... basic of that the idea ..." really needs some cleanup. Perhaps "The basis of the outlier detection plan for IFU data ..."

File: docs/jwst/outlier_detection/outlier_detection_ifu.rst [hunk 3]
```diff
@@ -14,7 +14,9 @@ in the :ref:`calwebb_detector1 <calwebb_detector1>` pipeline.
...
+that are regularly discrepant from their neighbors, with a sharper division than could be explained
+by the detector PSF. The algorithm flags pixels that are  outliers when compared to their neighbors for a set of
 input files contained in an association. The neighbor pixel differences are the neighbors in spatial direction. For MIRI data ,the neighbor differences are found to the left and right of every
 science pixel. While for NIRSpec data neighbor differences are
 found  between the pixels above and below every science pixel. The pixel differences for each input model
```
Comment (hbushouse) [inline docs/jwst/outlier_detection/outlier_detection_ifu.rst:22]:
```suggestion
found  between the pixels above and below every science pixel, due to differences in the dispersion direction for MIRI MRS and NIRSpec IFU images. The pixel differences for each input model 
```

File: jwst/outlier_detection/outlier_detection_ifu.py [hunk 4]
```diff
@@ -41,8 +41,10 @@ class OutlierDetectionIFU(OutlierDetection):
          a. read in science data
          b. Store computed neighbor differences for all the pixels.
             The neighbor pixel  differences are defined by the dispersion axis.
-            For MIRI (disp axis = 1) the neighbors to find differences  are to the left and right of pixel
-            For NIRSpec (disp axis = 0) the neighbors to find the differences are above and below the pixel
+            For MIRI (disp axis = 1, y axis) the neighbors to find differences are to 
+               the left and right of pixel
+            For NIRSpec (disp axis = 0, x axis) the neighbors to find the differences are 
+               above and below the pixel
```
Comment (hbushouse) [inline jwst/outlier_detection/outlier_detection_ifu.py:47]:
```suggestion
            For MIRI (disp axis = 2, y axis) the neighbors that are used to find differences are located to 
               the left and right of the pixel being examined
            For NIRSpec (disp axis = 1, x axis) the neighbors that are used to find differences are located
               above and below the pixel being examined
```

File: jwst/outlier_detection/outlier_detection_ifu.py [hunk 5]
```diff
@@ -206,8 +208,8 @@ def flag_outliers(self, idet, uq_det, ndet_files,
                 # set all science data that have DO_NOT_USE to NAN
                 sci[bad] = np.nan
 
-                # Compute left and right differences (MIRI dispersion axis = 1)
-                # For NIRSpec dispersion axis = 0, these differences are top, bottom
+                # Compute left and right differences (MIRI dispersion axis = 1 along y axis)
+                # For NIRSpec dispersion axis = 0 (along the x axis), these differences are top, bottom
```
Comment (hbushouse) [inline jwst/outlier_detection/outlier_detection_ifu.py:212]:
```suggestion
                # Compute left and right differences (MIRI dispersion axis = 2, along y axis)
                # For NIRSpec dispersion axis = 1 (along the x axis), these differences are top, bottom
```
The "dispaxis" (or whatever the right name is) keyword values in the header are 1-indexed, not 0-indexed.

**Comment by hbushouse** [comment]
@jemorrison If you can address the review comments we can get this merged.

**Comment by jemorrison** [comment]
@hbushouse Changes from review have been made

File: docs/jwst/outlier_detection/outlier_detection_ifu.rst [hunk 6]
```diff
@@ -14,11 +14,16 @@ in the :ref:`calwebb_detector1 <calwebb_detector1>` pipeline.
...
 
-An algorithm was developed to flag pixels that are  outliers when compared to their neighbors for a set of
-input files contained in an association. The neighbor pixel differences are the neighbors in spatial direction. For MIRI data ,the neighbor differences are found to the left and right of every
-science pixel. While for NIRSpec data neighbor differences are
-found  between the pixels above and below every science pixel. The pixel differences for each input model 
-in the association is determined and is stored in a stack of pixel differences. 
+The basis of the outlier detection flagging for IFU data  is to look for pixels on the detector
+that are regularly discrepant from their neighbors, with a sharper division than could be explained
+by the detector PSF. The algorithm flags pixels that are  outliers when compared to their neighbors for a set of
+input files contained in an association. The neighbor pixel differences are the neighbors in spatial direction.
```
Comment (hbushouse) [inline docs/jwst/outlier_detection/outlier_detection_ifu.rst:20]:
```suggestion
input files contained in an association. The neighbor pixel differences are the neighbors in the spatial direction.
```

File: jwst/outlier_detection/outlier_detection_ifu.py [hunk 7]
```diff
@@ -41,8 +41,10 @@ class OutlierDetectionIFU(OutlierDetection):
          a. read in science data
          b. Store computed neighbor differences for all the pixels.
             The neighbor pixel  differences are defined by the dispersion axis.
-            For MIRI (disp axis = 1) the neighbors to find differences  are to the left and right of pixel
-            For NIRSpec (disp axis = 0) the neighbors to find the differences are above and below the pixel
+            For MIRI, with the dispersion axis along the y axis, the neighbors that are used to
+            to find the differences are to the left and right of pixel being examined.
```
Comment (hbushouse) [inline jwst/outlier_detection/outlier_detection_ifu.py:45]:
```suggestion
            to find the differences are to the left and right of each pixel being examined.
```

**Review by hbushouse — APPROVED** [review]

**Pull request merged**